### PR TITLE
Extract runnable component

### PR DESCRIPTION
**Pull request by jnovack**
If this module is supposed to be a module, the use of this module (running it from the command line) should be extracted from the main code.

### CODE REVIEW

**Comment by jnovack** [comment]
I understand that this no longer makes it a "faithful" or "automated" port, however, it was wrong to begin with, we should make it right.

The application layer should be extracted from the processing layer.

**Comment by jnovack** [comment]
Rebased.

This is actually holding me up, as I have a number of additional changes in branches that use this as a base.

**Comment by Farof** [comment]
Can you update everything in `bin/`, `examples/` and `lib/`to work and reflect the change in the Readme examples? Thanks.

File: bin/heroprotocol.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+const heroprotocol = require('../');
+
+const HEADER            = heroprotocol.HEADER            = 'header';
+const DETAILS           = heroprotocol.DETAILS           = 'replay.details';
+const INITDATA          = heroprotocol.INITDATA          = 'replay.initdata';
+const GAME_EVENTS       = heroprotocol.GAME_EVENTS       = 'replay.game.events';
+const MESSAGE_EVENTS    = heroprotocol.MESSAGE_EVENTS    = 'replay.message.events';
+const TRACKER_EVENTS    = heroprotocol.TRACKER_EVENTS    = 'replay.tracker.events';
+const ATTRIBUTES_EVENTS = heroprotocol.ATTRIBUTES_EVENTS = 'replay.attributes.events';
```
Comment (Farof) [inline bin/heroprotocol.js]: Use `heroprotocol.HEADER` and similar instead of duplicating.

File: index.js [hunk 2]
```diff
@@ -141,130 +141,4 @@ exports.get = function (archiveFile, archive) {
...
-      logger.log(archive.get(ATTRIBUTES_EVENTS));
-    }
-
-    if (args.stats) {
-      logger.logStats();
-    }
-
-  })();
-}
+};
```
Comment (Farof) [inline index.js]: Keep the end of file empty line. Maybe update your editor preference so it doesn't remove it automatically on save.

File: lib/decoders.js [hunk 3]
```diff
@@ -417,4 +417,4 @@ VersionDecoder.prototype._skipInstance = function() {
   } else if (skip === 9) { // vint
     this._vint();
   }
-};
+};
```
Comment (Farof) [inline lib/decoders.js]: Same as above.

**Comment by jnovack** [comment]
As requested

**Pull request merged**